state-chain/pallets/cf-elections/src/electoral_systems/monotonic_median.rs

```diff
@@ -114,3 +114,127 @@ impl<
 		})
 	}
 }
+
+#[cfg(test)]
+mod test_monotonic_median {
+	use super::*;
+	use crate::electoral_system::{
+		mocks::MockElectoralSystem, ConsensusStatus, ElectoralReadAccess,
+	};
+
+	pub struct MockHook;
+
+	impl MedianChangeHook<u64> for MockHook {
+		fn on_change(_value: u64) {
+			HOOK_HAS_BEEN_CALLED.with(|hook_called| hook_called.set(true));
+		}
+	}
+
+	thread_local! {
+		pub static HOOK_HAS_BEEN_CALLED: std::cell::Cell<bool> = const { std::cell::Cell::new(false) };
+	}
+
+	const INIT_UNSYNCHRONISED_STATE: u64 = 22;
+
+	#[test]
+	fn check_consensus_correctly_calculates_median_when_all_authorities_vote() {
+		let mut electoral_system = MockElectoralSystem::<MonotonicMedian<u64, (), MockHook>>::new(
+			INIT_UNSYNCHRONISED_STATE,
+			(),
+			(),
+		);
+		let mut votes = vec![((), 1), ((), 5), ((), 3), ((), 2), ((), 8), ((), 6), ((), 9)];
```

Comment:
We can make this easier to read - the order of the values doesn't need to be randomised (it gets shuffled anyway). Also we could be more deliberate with the values we choose (see my other comment).